src/rpc/methods.zig

```diff
@@ -14,6 +19,320 @@ const Pubkey = sig.core.Pubkey;
 const Signature = sig.core.Signature;
 const Slot = sig.core.Slot;
 
+/// NOTE: for the sake of simplicity, we only support `method: ..., params: ...`,
+/// and reject `params: ..., method: ...`; this is a reasonable expectation for
+/// clients to satisfy.
+pub const Call = struct {
+    id: Id,
+    method: Method,
+
+    pub const Id = union(enum) {
+        null,
+        int: i128,
+        str: []const u8,
+    };
+
+    pub const Method = union(enum) {
```

Comment:
To me, this feels awkward as a nested struct. Could you imagine `Method` ever be used in a context where `Call` is not relevant?

Reply:
I suppose it currently is, I just nested it because it seemed like a logical way to structure it. I can move it outside.

Reply:
5dc22c6de2a6cd68930118671c3fa0bec4b08555